### DISCUSSION TITLE
How to read mysqldump into DuckDB?

### DISCUSSION THREAD

**Discussion by cdeil** [duckdb/duckdb#11751] [Q&A]
I have received a file `db.sql` that was created using [mysqldump](https://dev.mysql.com/doc/refman/8.0/en/mysqldump.html).

How can I get this data into DuckDB?

I tried `.read` (see [docs](https://duckdb.org/docs/api/cli/overview.html#reading-sql-from-a-file)) but got a ton of Parser Error.

Is it possible to read it somehow or convert it to something DuckDB can load?

Or is locally installing MySQL and then attaching and loading it from there (see [docs](https://duckdb.org/docs/guides/import/query_mysql.html)) the simplest option?

**Comment by cmdlineluser** [comment]
Have you tried [sqlglot](https://github.com/tobymao/sqlglot)?

```python
sqlglot.transpile(sql, read="mysql", write="duckdb", pretty=True)
```

**Comment by cdeil** [comment]
@cmdlineluser - thanks for the tip! Hadn't heard of `sqlglot` before.

The MySQL dump I have is 5 GB in size. I tried `sqlglot`, but then after 10 minutes killed the process when I saw memory increasing above 40 GB (and presumably wild SSD swapping going on).

I could try to locate and extract the relevant parts in the SQL file, but simply local MySQL restore and then attach & copy to DuckDB from there seems to be maybe the best route. I'll do that.
